core/src/main/scala/org/apache/spark/executor/TaskMetrics.scala

```diff
@@ -215,6 +215,18 @@ class TaskMetrics extends Serializable {
     inputMetrics.foreach(_.updateBytesRead())
   }
 
+  override def equals(other: Any): Boolean = other match {
```

Comment:
This is not a correct `equals`. There are other fields in `TaskMetrics`. To make `TaskMetrics.equals` work, there are also other classes need to be updated. Instead, could you just manually compare fields of DirectTaskResult in the tests rather than adding an incorrect `equals`?